community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/spi/QueryContext.scala

```diff
@@ -21,12 +21,12 @@ package org.neo4j.cypher.internal.compiler.v3_2.spi
 
 import java.net.URL
 
-import org.neo4j.cypher.internal.compiler.v3_2.InternalQueryStatistics
+import org.neo4j.cypher.internal.compiler.v3_2.{IndexDescriptor, InternalQueryStatistics}
 import org.neo4j.cypher.internal.compiler.v3_2.commands.expressions.{Expander, KernelPredicate, UserDefinedAggregator}
 import org.neo4j.cypher.internal.compiler.v3_2.pipes.matching.PatternNode
 import org.neo4j.cypher.internal.frontend.v3_2.SemanticDirection
 import org.neo4j.graphdb.{Node, Path, PropertyContainer, Relationship}
-import org.neo4j.cypher.internal.compiler.v3_2.IndexDescriptor
+import org.neo4j.io.pagecache.tracing.cursor.PageCursorTracer
```

Comment:
We should not introduce any new store dependencies here. Instead, create a trait in the compiler module, and implement it in the compatibility package. There you can have your kernel dependencies.